airflow/config_templates/config.yml

```diff
@@ -2041,6 +2041,13 @@ celery:
       sensitive: true
       example: "db+postgresql://postgres:airflow@postgres/airflow"
       default: ~
+    result_backend_engine_options:
```

Comment:
Does this only apply to the DB/SQLAlchemy backend as the description explains? If so we should change the name to be more specific (and if not we should change the description wording to be more generic).

Reply:
> Does this only apply to the DB/SQLAlchemy backend as the description explains? 

yes for the result backend db

> If so we should change the name to be more specific

something like `result_backend_sqlalchemy_engine_options`?

Reply:
Yeah that seems like a decent option! :+1: